test/options/techs.test.js

```diff
@@ -0,0 +1,139 @@
+const webpack = require('../helpers/compiler');
+
+describe('Options', () => {
+    describe('techs && techMap', () => {
+        test('only ts', async () => {
+            const mock = {
+                'index.ts' : `require('b:button')`,
+                'common.blocks' : {
+                    'button' : {
+                        'button.ts' : `({ block: 'button' })`
+                    }
+                },
+                'desktop.blocks' : {
+                    'button' : {
+                        'button.ts' : `({ block: 'button', content: 'desktop' })`
+                    }
+                }
+            };
+            const config = {
+                loader : {
+                    test : /\.ts$/,
+                    options : {
+                        levels : ['common.blocks', 'desktop.blocks'],
+                        // this is default
+                        techs : ['js'],
+                        techMap : {
+                            // to work with bem-react-core
+                            // we need to map js to ts
+                            js : ['ts']
+                        }
+                    }
+                }
+            };
+
+            const { stats } = await webpack('index.ts', { config, mock });
+            const { source } = stats.toJson().modules[1];
+
+            expect(source).toMatchSnapshot();
```

Comment:
Do you really like how this happens?
When you reading tests you don't see what's going on. You have to go to the some file named as concatenation of `e('Options', () => { describe('techs && techMap', () => { test('only ts', ` and then go back to compare with code. It's pretty annoying and uncomfortable.

Reply:
actually, I agree with you. But it's you know jest/hype snapshot testing. And I took an example from guys who write webpack its self.
Do you think it's better to assert strings ?